pkg/webhooks/jobset_webhook.go

```diff
@@ -308,17 +252,80 @@ func (j *jobSetWebhook) ValidateUpdate(ctx context.Context, old, newObj runtime.
 		return nil, fmt.Errorf("expected a JobSet from old object but got a %T", old)
 	}
 	mungedSpec := js.Spec.DeepCopy()
+
+	// Allow pod template to be mutated for suspended JobSets.
+	// This is needed for integration with Kueue/DWS.
 	if ptr.Deref(oldJS.Spec.Suspend, false) {
 		for index := range js.Spec.ReplicatedJobs {
-			mungedSpec.ReplicatedJobs[index].Template.Spec.Template.Spec.NodeSelector = oldJS.Spec.ReplicatedJobs[index].Template.Spec.Template.Spec.NodeSelector
+			mungedSpec.ReplicatedJobs[index].Template.Spec.Template = oldJS.Spec.ReplicatedJobs[index].Template.Spec.Template
```

Comment:
Question for @alculquicondor do Kueue/DWS require the ability to mutate any aspect of a pod template in a suspended JobSet? Or can we make this more targeted?

Reply:
Anything described here:
https://github.com/kubernetes-sigs/kueue/blob/a50d395c36a2cb3965be5232162cf1fded1bdb08/apis/kueue/v1beta1/workload_types.go#L256-L260

Reply:
Thanks, I updated the PR to only allow the fields in the PodSetUpdate struct to be mutated on JobSet updates.